ckanext/dcatapchharvest/profiles.py

```diff
@@ -796,18 +794,27 @@ def graph_from_dataset(self, dataset_dict, dataset_ref):  # noqa
                 ('media_type', DCAT.mediaType, None, Literal),
                 ('spatial', DCT.spatial, None, Literal),
             ]
-
             rights_uri = dh.get_license_uri_by_name(
                 resource_dict.get('rights')
+            )
+            if rights_uri is not None:
+                rights_ref = URIRef(rights_uri)
+                g.add((rights_ref, RDF.type, DCT.RightsStatement))
+                g.add((distribution, DCT.rights, rights_ref))
+            if rights_uri is None:
+                rights_name = dh.get_license_name_by_uri(
+                    resource_dict.get('rights')
                 )
-            g.add((rights_uri, RDF.type, DCT.RightsStatement))
-            g.add((dataset_ref, DCT.rights, rights_uri))
+            if rights_uri is None and rights_name is not None:
+                g.add((Literal(rights_name), RDF.type, DCT.RightsStatement))
```

Comment:
This shouldn't be a Literal, but a URIRef with the uri that was in the rights field of the resource.